### PR TITLE
WIP: Integrate with Middleware Endpoints

### PR DESCRIPTION
**Pull request by ej3076**

### CODE REVIEW

File: frontend/src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -16,10 +16,11 @@ export class AuthService {
   private _idToken: string;
   private _accessToken: string;
   private _expiresAt: number;
-  user: User;
+  private clientId: string = 't3sXyFtDUl0wFsHVsQsJbEa4en4bgPly';
```
Comment (ej3076) [inline frontend/src/app/auth/auth.service.ts:19]: As a nice-to-have, moving this out of the codebase would be nice

File: frontend/src/app/new-claim/new-claim.component.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            <ly-hint [align]="'after'"></ly-hint>
+            <ly-error *ngIf="color.hasError('required')"
+              >Color of vehicle is required</ly-error
+            >
+          </ly-field>
+        </ly-grid>
+
+        <ly-grid item [col]="'6 12@XSmall'">
+          <button ly-button bg="primary" raised (click)="sendNewVehicle()">
+            <span [className]="classes.labelBefore">Create Claim</span>
```
Comment (ej3076) [inline frontend/src/app/new-claim/new-claim.component.html:53]: Will put validation on the button here to ensure all fields are filled out with valid data

File: frontend/src/app/new-claim/new-claim.component.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+@Component({
+  selector: 'app-new-claim',
+  templateUrl: './new-claim.component.html',
+  styleUrls: ['./new-claim.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NewClaimComponent implements OnInit {
+  appearance = new FormControl();
+
+  vin = new FormControl('', [Validators.required, Validators.minLength(11)]);
```
Comment (sloan-seamus) [inline frontend/src/app/new-claim/new-claim.component.ts:22]: Aren't VIN's 17 characters/digits? Why not make the min length 17?

File: frontend/src/app/middleware/middleware.service.ts [hunk 4]
```diff
@@ -1,5 +1,9 @@
 import { Injectable } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Vehicle } from '../models/vehicle';
```
Comment (sloan-seamus) [inline frontend/src/app/middleware/middleware.service.ts:3]: Vehicle needs to be capitalized

File: frontend/src/app/auth/auth.service.ts (hunk 1 above)
Comment (dsifford) [reply frontend/src/app/auth/auth.service.ts:19]: Why not just use environment variables? Cant angular inject them?

**Pull request closed without merging**